File: chroma-league/src/main/java/com/bonepl/chromaleague/hud/parts/AssistKillingSpreeBar.java
package com.bonepl.chromaleague.hud.parts;

import com.bonepl.chromaleague.state.GameStateHelper;
import com.bonepl.razersdk.color.Color;
import com.bonepl.razersdk.animation.LayeredFrame;
import com.bonepl.razersdk.animation.SimpleFrame;
import com.bonepl.razersdk.color.StaticColor;
import com.bonepl.razersdk.sdk.RzKey;

import java.util.List;

import static com.bonepl.razersdk.sdk.RzKey.*;

public class AssistKillingSpreeBar extends LayeredFrame {
    private static final List<RzKey> KILLING_SPREE_BAR =
            List.of(RZKEY_M, RZKEY_COMA, RZKEY_DOT,
                    RZKEY_K, RZKEY_L, RZKEY_SEMICOLON,
                    RZKEY_O, RZKEY_P, RZKEY_SQUARE_BRACKET_LEFT);

    public AssistKillingSpreeBar() {
        addFrame(new SimpleFrame(KILLING_SPREE_BAR.subList(0, computeAssistsIndex()), StaticColor.YELLOW));
        addFrame(new SimpleFrame(KILLING_SPREE_BAR.subList(0, computeKillsIndex()), StaticColor.RED));
    }

    private static int computeAssistsIndex() {
        return Math.min(GameStateHelper.getActivePlayerAssistSpree(), KILLING_SPREE_BAR.size());
    }

    private static int computeKillsIndex() {
        return Math.min(GameStateHelper.getActivePlayerKillingSpree(), KILLING_SPREE_BAR.size());
    }

    public static List<RzKey> getKillingSpreeBar() {
        return KILLING_SPREE_BAR;
    }
}

File: src/hud/parts/AssistKillingSpreeBar.ts
import {LayeredFrame} from '../../razer-sdk/animation/LayeredFrame.js';
import {SimpleFrame} from '../../razer-sdk/animation/SimpleFrame.js';
import {StaticColor} from '../../razer-sdk/color/StaticColor.js';
import {RzKey} from '../../razer-sdk/sdk/RzKey.js';
import * as GameStateHelper from '../../state/GameStateHelper.js';

const KILLING_SPREE_BAR: RzKey[] = [
    RzKey.RZKEY_M, RzKey.RZKEY_COMA, RzKey.RZKEY_DOT,
    RzKey.RZKEY_K, RzKey.RZKEY_L, RzKey.RZKEY_SEMICOLON,
    RzKey.RZKEY_O, RzKey.RZKEY_P, RzKey.RZKEY_SQUARE_BRACKET_LEFT,
];

export class AssistKillingSpreeBar extends LayeredFrame {
    constructor() {
        super();
        this.addFrame(new SimpleFrame(KILLING_SPREE_BAR.slice(0, computeAssistsIndex()), StaticColor.YELLOW));
        this.addFrame(new SimpleFrame(KILLING_SPREE_BAR.slice(0, computeKillsIndex()), StaticColor.RED));
    }
}

function computeAssistsIndex(): number {
    return Math.min(GameStateHelper.getActivePlayerAssistSpree(), KILLING_SPREE_BAR.length);
}

function computeKillsIndex(): number {
    return Math.min(GameStateHelper.getActivePlayerKillingSpree(), KILLING_SPREE_BAR.length);
}

export function getKillingSpreeBar(): RzKey[] {
    return [...KILLING_SPREE_BAR];
}
